Sources/CoreMetrics/SystemMetrics.swift

```diff
@@ -0,0 +1,332 @@
+    ///
+    /// Backend implementations are encouraged to provide a static extension with
+    /// defaults that suit their specific backend needs.
+    public struct Labels {
+        /// Prefix to prefix all other labels with.
+        let prefix: String
+        /// Virtual memory size in bytes.
+        let virtualMemoryBytes: String
+        /// Resident memory size in bytes.
+        let residentMemoryBytes: String
+        /// Total user and system CPU time spent in seconds.
+        let startTimeSeconds: String
+        /// Total user and system CPU time spent in seconds.
+        let cpuSecondsTotal: String
+        /// Maximum number of open file descriptors.
+        let maxFileDescriptors: String
+        /// Number of open file descriptors.
+        let openFileDescriptors: String
+
+        /// Create a new `Labels` instance.
+        ///
+        /// - parameters:
+        ///     - prefix: Prefix to prefix all other labels with.
+        ///     - virtualMemoryBytes: Virtual memory size in bytes
+        ///     - residentMemoryBytes: Resident memory size in bytes.
+        ///     - startTimeSeconds: Total user and system CPU time spent in seconds.
+        ///     - cpuSecondsTotal: Total user and system CPU time spent in seconds.
+        ///     - maxFds: Maximum number of open file descriptors.
+        ///     - openFds: Number of open file descriptors.
+        public init(prefix: String, virtualMemoryBytes: String, residentMemoryBytes: String, startTimeSeconds: String, cpuSecondsTotal: String, maxFds: String, openFds: String) {
+            self.prefix = prefix
+            self.virtualMemoryBytes = virtualMemoryBytes
+            self.residentMemoryBytes = residentMemoryBytes
+            self.startTimeSeconds = startTimeSeconds
+            self.cpuSecondsTotal = cpuSecondsTotal
+            self.maxFileDescriptors = maxFds
+            self.openFileDescriptors = openFds
+        }
+
+        func label(for keyPath: KeyPath<Labels, String>) -> String {
+            return self.prefix + self[keyPath: keyPath]
+        }
+    }
+
+    /// System Metric data.
+    ///
+    /// The current list of metrics exposed is taken from the Prometheus Client Library Guidelines
+    /// https://prometheus.io/docs/instrumenting/writing_clientlibs/#standard-and-runtime-collectors
+    public struct Data {
+        /// Virtual memory size in bytes.
+        var virtualMemoryBytes: Int
+        /// Resident memory size in bytes.
+        var residentMemoryBytes: Int
+        /// Start time of the process since unix epoch in seconds.
+        var startTimeSeconds: Int
+        /// Total user and system CPU time spent in seconds.
+        var cpuSeconds: Int
+        /// Maximum number of open file descriptors.
+        var maxFileDescriptors: Int
+        /// Number of open file descriptors.
+        var openFileDescriptors: Int
+    }
+
+    #if os(Linux)
+    internal static func linuxSystemMetrics() -> SystemMetrics.Data? {
```

Comment:
@drexin @ktoso can you help review the part

Reply:
I would suggest using Foundation's `FileManager` instead, which would boil this down to

```swift
guard let statData = FileManager.default.contents(atPath: "/proc/self/stat") else {
  return nil
}

guard let stats = String(data: statData, encoding: .utf8) else {
  return nil
}
```

Reply:
We used to have an implementation based on `FileManager`, but per @tomerd's request we attempted to keep `CoreMetrics` 100% Foundation free, which is why we went with the C approach.